### PR TITLE
Frontend client internal types

### PR DESCRIPTION
**Pull request by vytautas-karpavicius**
**What changed?**
Switch frontend client to internal types. Changed immediate client usage by passing already constructed types or using thrift mappers at the call boundary. Upcoming PRs will propagate those conversions further up, until internal types reaches handlers eventually.

**Why?**
To propagate internal types throughout Cadence codebase.

**How did you test it?**
Existing tests.

**Potential risks**

### CODE REVIEW

File: service/frontend/dcRedirectionHandler_test.go [hunk 1]
```diff
@@ -24,12 +24,15 @@ import (
 	"context"
 	"testing"
 
+	"github.com/uber/cadence/client/frontend"
+	"github.com/uber/cadence/common/types"
+	"github.com/uber/cadence/common/types/mapper/thrift"
```
Comment (yycptt) [inline service/frontend/dcRedirectionHandler_test.go:29]: nit: please group with other cadence imports

**Review by yycptt — APPROVED** [review]

File: service/worker/batcher/workflow.go [hunk 2]
```diff
@@ -367,27 +367,27 @@ func startTaskProcessor(
 				err = processTask(ctx, limiter, task, batchParams, client,
 					batchParams.CancelParams.CancelChildren,
 					func(workflowID, runID string) error {
-						return client.RequestCancelWorkflowExecution(ctx, &shared.RequestCancelWorkflowExecutionRequest{
+						return client.RequestCancelWorkflowExecution(ctx, &types.RequestCancelWorkflowExecutionRequest{
```
Comment (yycptt) [inline service/worker/batcher/workflow.go:370]: I am not entirely sure if we need to convert the error. Seems like EntityNotExist is the only error we care about and it's already updated? Same for terminate and signal?

File: service/frontend/dcRedirectionHandler_test.go (hunk 1 above)
Comment (vytautas-karpavicius) [reply service/frontend/dcRedirectionHandler_test.go]: Done.

File: service/worker/batcher/workflow.go (hunk 2 above)
Comment (vytautas-karpavicius) [reply service/worker/batcher/workflow.go:370]: We don't need that here. Both batcher and failovermanager workers were simple enough to update in one shot. Thus no thrift related types (including errors) left there.
This particular error returned here is checked against `*types.EntityNotExistsError` just as you mentioned, which is already converted as well.

**Pull request merged**